workspaces/global-floating-action-button/plugins/global-floating-action-button/src/utils.ts

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2025 The Backstage Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import {
+  FloatingActionButton,
+  FloatingActionButtonWithPositions,
+  Slot,
+} from './types';
+
+export const evaluateFloatingButtonsWithPositions = (
+  floatingButtons: FloatingActionButton[],
+): FloatingActionButtonWithPositions =>
+  floatingButtons.reduce(
+    (acc: FloatingActionButtonWithPositions, fb: FloatingActionButton) => {
+      const position =
+        !fb?.position || !(fb.position in Slot)
+          ? Slot.PAGE_END
+          : Slot[fb.position as keyof typeof Slot];
+      if (position in acc) {
+        acc[position].push(fb);
+      } else {
+        acc[position] = [fb];
+      }
+      return acc;
+    },
+    {},
+  );
+
+export const sortButtonsWithPriority = (
+  floatingButtons: FloatingActionButton[],
+) => {
+  const buttons = [...floatingButtons];
+  return buttons.sort((fb1, fb2) => {
+    if ((fb2.priority || 0) > (fb1.priority || 0)) {
+      return 1;
+    }
+    if ((fb2.priority || 0) < (fb1.priority || 0)) {
+      return -1;
+    }
+    return 0;
+  });
+};
+
+export const filterAndSortButtons = (
+  floatingButtons: FloatingActionButton[],
+  pathname: string,
+) => {
+  const sortedButtons = sortButtonsWithPriority(floatingButtons);
+  return sortedButtons.filter(fb => {
```

Comment:
Just a nit and I'm sure you can't measure it with our 3 floating buttons. But I would always first filter and then sort elements to reduce the number of sorting elements. :man_shrugging: